source/OptimaJet.DWKit.StarterApplication/Controllers/BaseController.cs

```diff
@@ -35,13 +36,14 @@ public User CurrentUser
         {
             get {
                 var exists = HttpContext.Items.ContainsKey(CURRENT_USER_KEY);
+                var existing = HttpContext.Items[CURRENT_USER_KEY];
 
-                if (exists) return (User)HttpContext.Items[CURRENT_USER_KEY];
+                if (exists && existing != null) return (User)existing;
 
                 // current user has not yet been found for this request.
                 // find or create because users are managed by auth0 and
                 // creation isn't proxied through the api.
-                var user = FindOrCreateCurrentUser();
+                var user = FindOrCreateCurrentUser().Result;
```

Comment:
since async/await is more of a natural flow of async behavior, I pulled of the sync forcing up a level to here